libs/num/numeric.go

```diff
@@ -0,0 +1,83 @@
+package num
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+type Numeric struct {
+	AsUint    *Uint
```

Comment:
I would not export these fields. This is prone to abuse and concurrency issues/bugs due to shared pointers

Reply:
Addressed.
Changed everywhere to set and get via methods.
Thanks!